### PR TITLE
Wire up trending tokens UI

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-2049
Fixes APP-1960

## What changed (plus any additional context for devs)
Wires up the react-query hook and displays the trending token list for the selected network (or all if none). Also creates two new Zustand stores: trendingTokens, and networkSwitcher which hold appropriate data to make the trending tokens and network switcher work well.

Also did some minor shuffling of files here as I noticed that we were having an ever-growing `components` folder inside of the `screens/discover` folder. I moved those to their own `Discover` folder inside of top-level `components`.

## Screen recordings / screenshots

https://github.com/user-attachments/assets/97d18cb0-2c79-4ba7-9c76-02e57ce2285d

## What to test
Test each network and see that a list of tokens are displayed or no results if nothing is on that chain (I think that's the case for Apechain?)

### CODE REVIEW

File: src/helpers/strings.ts
Comment (walmat) [inline src/helpers/strings.ts:1]: not sure if there's a better way to display this shorthand notation in the figma

File: src/hooks/reanimated/useSyncSharedValue.ts
Comment (walmat) [inline src/hooks/reanimated/useSyncSharedValue.ts:1]: @christianbaroni these `| undefined` were preventing me from syncing to `undefined` values

Comment (christianbaroni) [reply src/hooks/reanimated/useSyncSharedValue.ts:1]: Should be fine to make this change

File: src/screens/discover/components/NetworkSwitcher.tsx [hunk 1]
```diff
@@ -746,28 +755,24 @@ function Sheet({ children, header, onClose }: PropsWithChildren<{ header: ReactE
...
+  const network = useTrendingTokensStore(state => state.network);
+  const selected = useSharedValue<ChainId | undefined>(network);
+  const setNetwork = useTrendingTokensStore(state => state.setNetwork);
+
+  useSyncSharedValue<ChainId | undefined>({
+    sharedValue: selected,
+    syncDirection: 'sharedValueToState',
+    state: network,
+    setState: setNetwork,
+  });
```
Comment (greg-schrammel) [inline src/screens/discover/components/NetworkSwitcher.tsx:770]: I'm not against doing this here, but just for context on the tradeoff, every selection change is gonna rerender/fetch the trending tokens screen behind the wallet switcher

so close and already very likely have the updated list data, but do more work while selecting 
vs 
only update the list when it's visible

File: src/screens/discover/components/NetworkSwitcher.tsx [hunk 2]
```diff
@@ -746,28 +755,24 @@ function Sheet({ children, header, onClose }: PropsWithChildren<{ header: ReactE
...
   const editing = useSharedValue(false);
-  const selected = useSharedValue<ChainId[] | 'all'>([]);
 
-  const close = () => {
-    'worklet';
-    runOnJS(onClose)(selected.value);
-  };
+  const network = useTrendingTokensStore(state => state.network);
+  const selected = useSharedValue<ChainId | undefined>(network);
+  const setNetwork = useTrendingTokensStore(state => state.setNetwork);
```
Comment (greg-schrammel) [inline src/screens/discover/components/NetworkSwitcher.tsx:763]: I'm pretty sure this component is gonna be used in other places later, fine for now but lets try not to couple it too much

File: src/screens/discover/components/NetworkSwitcher.tsx [hunk 3]
```diff
@@ -620,29 +615,36 @@ function NetworksGrid({ editing, selected }: { editing: SharedValue<boolean>; se
...
-    const chainId = networks.value[section][index];
-
-    selected.modify(selected => {
-      if (selected === 'all') {
-        // @ts-expect-error I think something is wrong with reanimated types, not infering right here
-        selected = [chainId];
-        return selected;
+  const pressedExpand = useSharedValue(false);
+
+  // TODO: Need to prevent this from firing the tapNetwork as well
```
[the remark this answers is not in the document]
Comment (greg-schrammel) [reply src/screens/discover/components/NetworkSwitcher.tsx:620]: why is that?

File: src/__swaps__/screens/Swap/components/SwapCoinIcon.tsx [hunk 4]
```diff
@@ -68,25 +53,21 @@ export const SwapCoinIcon = React.memo(function FeedCoinIcon({
...
   iconUrl?: string;
   disableShadow?: boolean;
   forceDarkMode?: boolean;
-  large?: boolean;
   mainnetAddress?: string;
   chainId: ChainId;
-  small?: boolean;
   symbol: string;
-  xLarge?: boolean;
+  size?: number;
```
[the remark this answers is not in the document]
Comment (walmat) [reply src/__swaps__/screens/Swap/components/SwapCoinIcon.tsx:70]: no it defaults to `32`

[the remark this answers is not in the document]
Comment (benisgold) [reply src/__swaps__/screens/Swap/components/SwapCoinIcon.tsx:70]: oh

**Review by benisgold — APPROVED** [review]

File: src/screens/discover/components/NetworkSwitcher.tsx (hunk 2 above)
Comment (walmat) [reply src/screens/discover/components/NetworkSwitcher.tsx]: decoupled it, thanks for the notice on this

File: src/screens/discover/components/NetworkSwitcher.tsx (hunk 1 above)
Comment (walmat) [reply src/screens/discover/components/NetworkSwitcher.tsx]: removed this also, ty for the tip

**Review by greg-schrammel — APPROVED** [review]

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/wire-up-trending-tokens/062ac0f0f22693770ae7be6a8e522c271b3dce29.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/wire-up-trending-tokens/062ac0f0f22693770ae7be6a8e522c271b3dce29.ipa) for 062ac0f0f22693770ae7be6a8e522c271b3dce29

**Pull request merged**